packages/bundle-source/src/fs.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @param {string} fileName
+ * @param {{
+ *   fs: {
+ *     promises: Pick<import('fs/promises'),'readFile' | 'stat'>
+ *   },
+ *   path: Pick<import('path'), 'resolve' | 'relative' | 'normalize'>,
+ * }} powers
+ */
+export const makeFileReader = (fileName, { fs, path }) => {
+  const make = there => makeFileReader(there, { fs, path });
+
+  // fs.promises.exists isn't implemented in Node.js apparently because it's pure
+  // sugar.
+  const exists = fn =>
+    fs.promises.stat(fn).then(
+      () => true,
+      e => {
+        if (e.code === 'ENOENT') {
+          return false;
+        }
+        throw e;
+      },
+    );
```

Comment:
This is now dead code but I figure we’ll want to decide whether to keep it after we start building out a public `@endo/fs` API.